test/parallel/test-cluster-worker-wait-server-close.js

```diff
@@ -22,10 +22,12 @@ if (cluster.isWorker) {
   worker.once('listening', function() {
     net.createConnection(common.PORT, common.localhostIPv4, function() {
       var socket = this;
-      worker.disconnect();
-      setTimeout(function() {
-        socket.write('.');
-        connectionDone = true;
+      setTimeout(function(){
```

Comment:
@Olegas I didn't say it was imaginary! I said that inserting arbitrary timeouts to solve undescribed race conditions is poor practice. You didn't answer my question: what is the race condition?

Can you describe what ordering of scheduling events is causing this to break on OS X? Usually, rather than timeouts, its possible to explicitly wait for precise events to avoid race conditions and timeouts. If node lacks such events, then timeouts are our *last* resort.